controller/api/public/stat_summary.go

```diff
@@ -0,0 +1,311 @@
+	}
+
+	rsp := pb.StatSummaryResponse{
+		Response: &pb.StatSummaryResponse_Ok_{ // https://github.com/golang/protobuf/issues/205
+			Ok: &pb.StatSummaryResponse_Ok{
+				StatTables: []*pb.StatTable{
+					&pb.StatTable{
+						Table: &pb.StatTable_PodGroup_{
+							&pb.StatTable_PodGroup{
+								Rows: rows,
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	return &rsp, nil
+}
+
+func promLabel(key, val string) string {
+	return fmt.Sprintf("%s=\"%s\"", key, val)
+}
+
+func buildRequestLabels(req *pb.StatSummaryRequest) string {
+	labels := []string{
+		promLabel("namespace", req.Resource.Spec.Namespace),
+	}
+
+	if req.Resource.Spec.Name != "" {
+		labels = append(labels, promLabel(req.Resource.Spec.Type, req.Resource.Spec.Name))
+	}
+
+	var direction string
+	switch req.Outbound.(type) {
+	case *pb.StatSummaryRequest_None:
+		direction = "inbound"
+	case *pb.StatSummaryRequest_OutToResource:
+		direction = "outbound"
+
+		dstLabel := fmt.Sprintf("dst_namespace=\"%s\", dst_%s=\"%s\"",
+			req.GetOutToResource().Namespace,
+			req.GetOutToResource().Type,
+			req.GetOutToResource().Name,
+		)
+		labels = append(labels, dstLabel)
+	case *pb.StatSummaryRequest_OutFromResource:
+		direction = "outbound"
+
+		srcLabel := fmt.Sprintf("src_namespace=\"%s\", src_%s=\"%s\"",
+			req.GetOutFromResource().Namespace, // are we going to enforce that ns/type/name are all set?
+			req.GetOutFromResource().Type,
+			req.GetOutFromResource().Name,
+		)
+		labels = append(labels, srcLabel)
+	default:
+		direction = "inbound"
+	}
+	labels = append(labels, promLabel("direction", direction))
+
+	return strings.Join(labels, ",")
+}
+
+func (h *handler) getRequests(ctx context.Context, reqLabels string, groupBy string, timeWindow string) (map[string]map[string]uint64, error) {
+	requestsQuery := fmt.Sprintf(reqQuery, reqLabels, timeWindow, groupBy)
+
+	resultVector, err := h.QueryProm(ctx, requestsQuery)
+	if err != nil {
+		return nil, err
+	}
+	return processRequests(resultVector, groupBy), nil
+}
+
+func processRequests(vec model.Vector, labelSelector string) map[string]map[string]uint64 {
+	result := make(map[string]map[string]uint64, 0)
+	for _, sample := range vec {
+		labels := metricToMap(sample.Metric)
+		if result[labels[labelSelector]] == nil {
+			result[labels[labelSelector]] = make(map[string]uint64)
+		}
+
+		// 'increase' extrapolates values, so sample.Value could be a non-integer
+		result[labels[labelSelector]][labels["classification"]] = uint64(sample.Value)
+	}
+	return result
+}
+
+func metricToMap(metric model.Metric) map[string]string {
+	labels := make(map[string]string)
+	for k, v := range metric {
+		labels[string(k)] = string(v)
+	}
+	return labels
+}
+
+func (h *handler) getDeployment(namespace string, name string) ([]v1beta1.Deployment, map[string]meshedCount, error) {
+	if namespace == "" {
+		namespace = apiv1.NamespaceDefault
+	}
+
+	deploymentsClient := h.k8sClient.AppsV1beta1().Deployments(namespace)
```

Comment:
The version of client-go that we're using is based on the kuberntes 1.9 API, where apps have been graduated from V1beta1 to V1. So you can write this as:

```
	deploymentsClient := h.k8sClient.Apps().Deployments(namespace)
```